website/static/js/projectSettings.js

```diff
@@ -8,77 +8,114 @@ var $osf = require('js/osfHelpers');
-            self.UPDATE_SUCCESS_MESSAGE = 'Category updated successfully';
-            self.UPDATE_ERROR_MESSAGE = 'Error updating category, please try again. If the problem persists, email ' +
+            self.decodedTitle = params.currentTitle;
+            self.decodedDescription = $osf.htmlDecode(params.currentDescription);
+            self.title = ko.observable(params.currentTitle).extend({
+                required: {
+                    params: true,
+                    message: 'Title cannot be blank.'
+                }});
+            self.description = ko.observable(self.decodedDescription);
+
+            self.disabled = params.disabled || false;
+
+            self.UPDATE_SUCCESS_MESSAGE = 'Category, title, and description updated successfully';
+            self.UPDATE_CATEGORY_ERROR_MESSAGE = 'Error updating category, please try again. If the problem persists, email ' +
                 '<a href="mailto:support@osf.io">support@osf.io</a>.';
-            self.UPDATE_ERROR_MESSAGE_RAVEN = 'Error updating Node.category';
+            self.UPDATE_DESCRIPTION_ERROR_MESSAGE = 'Error updating description, please try again. If the problem persists, email ' +
+                '<a href="mailto:support@osf.io">support@osf.io</a>.';
+            self.UPDATE_CATEGORY_ERROR_MESSAGE_RAVEN = 'Error updating Node.category';
 
-            self.INSTANTIATION_ERROR_MESSAGE = 'Trying to instantiate NodeCategorySettings view model without an update URL';
+            self.INSTANTIATION_ERROR_MESSAGE = 'Trying to instantiate ProjectSettings view model without an update URL';
 
             self.MESSAGE_SUCCESS_CLASS = 'text-success';
             self.MESSAGE_ERROR_CLASS = 'text-danger';
 
-            if (!updateUrl) {
+            if (!params.updateUrl) {
                 throw new Error(self.INSTANTIATION_ERROR_MESSAGE);
             }
 
-            self.categories = categories;
-            self.category = ko.observable(category);
-            self.updateUrl = updateUrl;
-
-            self.selectedCategory = ko.observable(category);
-            self.dirty = ko.observable(false);
-            self.selectedCategory.subscribe(function(value) {
-                if (value !== self.category()) {
-                    self.dirty(true);
-                }
-            });
+            self.categories = params.categories;
+            self.category = params.category;
+            self.updateUrl = params.updateUrl;
+            self.selectedCategory = ko.observable(params.category);
         },
-        updateSuccess: function(newcategory) {
+
+        /*success handler*/
+        updateSuccess: function() {
             var self = this;
             self.changeMessage(self.UPDATE_SUCCESS_MESSAGE, self.MESSAGE_SUCCESS_CLASS);
-            self.category(newcategory);
-            self.dirty(false);
         },
-        updateError: function(xhr, status, error) {
+
+        /*error handlers*/
+        updateCategoryError: function(xhr, status, error) {
```

Comment:
One comment that Steve made to me as a standard going forward: to help write re-usable code, we'd ideally like a clean separation of concerns. Instead of a viewmodel does three things (stores data, communicates with the server, and updates the page/ flashes warnings), it's beneficial to have three separate objects/classes:

- An object that holds data
- An object that sends data back and forth to the server (a client)
- A small ViewModel that just coordinates and updates the UI

I won't pretend `apiApplication.js` is perfect, but it illustrates something along those lines of separation. I'd be happy to discuss this further if you wanted to think about how to break this down.